backend/pkg/commons/db2/raw/client.go

```diff
@@ -0,0 +1,239 @@
+package raw
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/gobitfly/beaconchain/pkg/commons/db2/database"
+	"github.com/gobitfly/beaconchain/pkg/commons/db2/jsonrpc"
+)
+
+var ErrNotFoundInCache = fmt.Errorf("cannot find hash in cache")
+var ErrMethodNotSupported = fmt.Errorf("method not supported")
+
+type StoreReader interface {
+	ReadBlockByNumber(chainID uint64, number int64) (*FullBlockData, error)
+	ReadBlockByHash(chainID uint64, hash string) (*FullBlockData, error)
+	ReadBlocksByNumber(chainID uint64, start, end int64) ([]*FullBlockData, error)
+}
+
+type WithFallback struct {
+	roundTripper http.RoundTripper
+	fallback     http.RoundTripper
+}
+
+func NewWithFallback(roundTripper, fallback http.RoundTripper) *WithFallback {
+	return &WithFallback{
+		roundTripper: roundTripper,
+		fallback:     fallback,
+	}
+}
+
+func (r WithFallback) RoundTrip(request *http.Request) (*http.Response, error) {
+	resp, err := r.roundTripper.RoundTrip(request)
+	if err == nil {
+		// no fallback needed
+		return resp, nil
+	}
+
+	var e1 *json.SyntaxError
+	if !errors.As(err, &e1) &&
+		!errors.Is(err, ErrNotFoundInCache) &&
+		!errors.Is(err, ErrMethodNotSupported) &&
+		!errors.Is(err, database.ErrNotFound) {
+		return nil, err
+	}
+
+	return r.fallback.RoundTrip(request)
+}
+
+type StoreRoundTripper struct {
+	store   StoreReader
+	chainID uint64
+}
+
+func NewBigTableEthRaw(store StoreReader, chainID uint64) *StoreRoundTripper {
+	return &StoreRoundTripper{
+		store:   store,
+		chainID: chainID,
+	}
+}
+
+func (r *StoreRoundTripper) RoundTrip(request *http.Request) (*http.Response, error) {
+	body, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		request.Body = io.NopCloser(bytes.NewBuffer(body))
+	}()
+	var messages []*jsonrpc.Message
+	var isSingle bool
```

Comment:
```suggestion
	var isSingleMsg bool
```
Maybe consider renaming this to `isSingleMsg` for better clarity